kahuna/public/js/search/results.js

```diff
@@ -19,6 +19,16 @@ function compact(array) {
     return array.filter(angular.isDefined);
 }
 
+
+// Global session-level state to remember the uploadTime of the first
+// result in the last search.  This allows to always paginate the same
+// set of results, as well as recovering the same set of results if
+// navigating back to the same search.
+// Note: I tried to do this using non-URL $stateParams and it was a
+// rabbit-hole that doesn't seem to have any end. Hence this slightly
+// horrid global state.
+let lastSearchFirstResultTime;
```

Comment:
Yes I thought so, but I couldn't really find any reason to justify the accompagnying boilerplate. Can you think of any?

One would be to help mock and write tests but we don't do that at the moment...